### PR TITLE
Refactored admin flash spec

### PR DESCRIPTION
**Pull request by KevinMulhern**
fixes https://github.com/TheOdinProject/theodinproject/issues/261

### CODE REVIEW

File: spec/models/lesson_completion_spec.rb [hunk 1]
```diff
@@ -1,19 +1,13 @@
 require 'rails_helper'
 
 describe LessonCompletion do
+  subject(:lesson_completion) {
+    LessonCompletion.new(:student_id => 1,:lesson_id => 1)
```
Comment (KevinMulhern) [inline spec/models/lesson_completion_spec.rb]: Updated 👍

File: app/models/admin_flash.rb [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 
   def self.unexpired_messages
-    self.order("created_at desc").where("expires >= ?", Time.now)
+    self.order('created_at desc').where('expires >= ?', Time.now)
   end
 
-  def self.showable_messages disabled_flash_ids
-    unexpired_messages.select{ |msg| !disabled_flash_ids.include?(msg.id) }
+  def self.showable_messages(disabled_flash_ids)
+    unexpired_messages.select do |message|
```
Comment (arun1595) [inline app/models/admin_flash.rb]: Can we use `reject` instead to get rid of `!` ?

IMO, it makes it a bit hard to understand the code

Comment (KevinMulhern) [inline app/models/admin_flash.rb]: Nice suggestion, I'll change it now.

File: spec/models/admin_flash_spec.rb [hunk 3]
```diff
@@ -1,75 +1,37 @@
...
-      expect { admin_flash.destroy }.to change { AdminFlash.count }.by(-1)
-    end
-  end
-
-  describe "When 'AdminFlash' message is updated" do
-    it 'returns true(update_attributes)' do
-      admin_flash = AdminFlash.create(message: "Lorem ipsum", expires: 2.days.from_now)
-      #assert true, admin_flash.update_attribute(:message => "Asdf Qwerty")
-      expect(admin_flash.update_attribute(:message, "Asdf Qwerty")).to be_truthy
+    it 'returns array ofs messages without cookies' do
```
Comment (arun1595) [inline spec/models/admin_flash_spec.rb]: A typo in 'of'

File: spec/models/admin_flash_spec.rb [hunk 4]
```diff
@@ -1,75 +1,37 @@
...
+      and_return([admin_flash])
   end
 
-  describe "#unexpired_messages" do
-    before do
-      admin_flash_2.save
-      admin_flash.save
-    end
-    # tests to make sure it only grabs active flash messages
+  describe "#.unexpired_messages" do
```
Comment (arun1595) [inline spec/models/admin_flash_spec.rb]: Lets get rid of "#" as it is a class method.

File: spec/models/admin_flash_spec.rb [hunk 5]
```diff
@@ -1,75 +1,37 @@
...
+    allow(AdminFlash).to receive(:order).with('created_at desc').
+      and_return(admin_flashes)
+
+    allow(Time).to receive(:now).and_return(current_time)
+
+    allow(admin_flashes).to receive(:where).with('expires >= ?', current_time).
+      and_return([admin_flash])
   end
 
   describe "#unexpired_messages" do
```
Comment (arun1595) [inline spec/models/admin_flash_spec.rb]: I think it should read `describe ".unexpired_messages" do` as we're describing a class method.

Comment (KevinMulhern) [inline spec/models/admin_flash_spec.rb]: Updated

**Pull request merged**